### PR TITLE
Added formula to install python aeneas

### PR DESCRIPTION
**Pull request by danielbair**

### CODE REVIEW

**Comment by tdsmith** [comment]
Thanks for the PR! I'll review this sometime this weekend.

File: aeneas.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
+class Aeneas < Formula
+  desc "Python/C library and set of tools to synchronize audio and text"
+  homepage "http://www.readbeyond.it/aeneas/"
+  url "https://github.com/readbeyond/aeneas/archive/v1.5.0.3.tar.gz"
+  sha256 "fa979e701f89440631afc474b2fb27f5fbf5702c582d424c8d9bb52dcec80fb4"
+  head "https://github.com/readbeyond/aeneas.git"
+
+  depends_on "ffmpeg"
+  depends_on "danielbair/tap/espeak"
```
Comment (tdsmith) [inline aeneas.rb]: Sorry, we can't accept formulas with dependencies on taps outside the Homebrew organization.

File: aeneas.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    url "https://pypi.python.org/packages/11/1b/fe6904151b37a0d6da6e60c13583945f8ce3eae8ebd0ec763ce546358947/lxml-3.6.0.tar.gz"
+    sha256 "9c74ca28a7f0c30dca8872281b3c47705e21217c8bc63912d95c9e2a7cac6bdf"
+  end
+
+  resource "numpy" do
+    url "https://pypi.python.org/packages/e0/4c/515d7c4ac424ff38cc919f7099bf293dd064ba9a600e1e3835b3edefdb18/numpy-1.11.1.tar.gz"
+    sha256 "dc4082c43979cc856a2bf352a8297ea109ccb3244d783ae067eb2ee5b0d577cd"
+  end
+
+  patch :DATA
```
Comment (tdsmith) [inline aeneas.rb]: Have these patches been submitted upstream?

File: aeneas.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+    system "python", *Language::Python.setup_install_args(libexec)
+
+    bin.install Dir["#{libexec}/bin/*"]
+    cp "VERSION", libexec/"lib/python2.7/site-packages"
+    cp "check_dependencies.py", libexec/"lib/python2.7/site-packages"
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+    `grep "#{libexec}/lib/python2.7/site-packages" /Users/$(whoami)/.bash_profile > /dev/null || echo "export PYTHONPATH=#{ENV["PYTHONPATH"]}:$PYTHONPATH" >> /Users/$(whoami)/.bash_profile`
+    `grep "/usr/local/bin:/usr/local/sbin" /Users/$(whoami)/.bash_profile > /dev/null || echo "export PATH=/usr/local/bin:/usr/local/sbin:$PATH" >> /Users/$(whoami)/.bash_profile`
```
Comment (tdsmith) [inline aeneas.rb]: Modifying files in the user's home directory from a formula is not acceptable, sorry!

File: aeneas.rb (hunk 2 above)
Comment (danielbair) [inline aeneas.rb]: Yes. I am working with the developer to accept these patches.

File: aeneas.rb (hunk 1 above)
Comment (danielbair) [inline aeneas.rb]: Ok, I have submitted a pull request for my espeak tap to homebrew-core and if it gets accepted I will update this formula. If not accepted I will work on plan b.

File: aeneas.rb (hunk 3 above)
Comment (danielbair) [inline aeneas.rb]: Sorry, I was not aware of this. I will remove the lines that update the users .bash_profile and put instructions for the user to do this themselves in the caveats section.

**Comment by danielbair** [comment]
@tdsmith would you be able to review this formula again? I have tried to implement the requirements you gave.

File: aeneas.rb [hunk 4]
```diff
@@ -0,0 +1,99 @@
+class Aeneas < Formula
+  desc "Python/C library and set of tools to synchronize audio and text"
+  homepage "http://www.readbeyond.it/aeneas/"
+  url "https://github.com/readbeyond/aeneas/archive/v1.5.0.3.tar.gz"
+  sha256 "fa979e701f89440631afc474b2fb27f5fbf5702c582d424c8d9bb52dcec80fb4"
+  head "https://github.com/readbeyond/aeneas.git"
+
+  option "with-libespeak", "Build with libespeak for aeneas.cew high speed synthesis. (To build you will need to first 'brew install danielbair/tap/espeak' to provide libespeak as the current homebrew-core espeak formula does not provide this.)"
```
Comment (tdsmith) [inline aeneas.rb]: Can the Homebrew-core formula be modified to provide this, at least optionally?

File: aeneas.rb [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    <<-EOS.undent
+
+        To install --with-libespeak you will need to first 'brew install danielbair/tap/espeak' to provide libespeak as the current homebrew-core espeak formula does not provide this.
+
+        If you use system python (that comes - depending on the OS X version - with older versions of numpy, scipy and matplotlib), you may need to ensure that the brewed packages come earlier in Python's sys.path with:
+          mkdir -p #{user_site_packages}
+          echo 'import sys; sys.path.insert(1, "#{homebrew_site_packages1}"); sys.path.insert(1, "#{homebrew_site_packages2}")' >> #{user_site_packages}/homebrew.pth
+
+    EOS
+  end
```
Comment (tdsmith) [inline aeneas.rb]: These caveats should not be necessary.

File: aeneas.rb [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+  def install
+    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
+    resources.each do |r|
+      r.stage do
+        system "python", *Language::Python.setup_install_args(libexec/"vendor")
+      end
+    end
+
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+    system "python", *Language::Python.setup_install_args(libexec)
```
Comment (tdsmith) [inline aeneas.rb]: If this is intended to be `import`able, you should just install into prefix.

File: aeneas.rb [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    url "https://pypi.python.org/packages/11/1b/fe6904151b37a0d6da6e60c13583945f8ce3eae8ebd0ec763ce546358947/lxml-3.6.0.tar.gz"
+    sha256 "9c74ca28a7f0c30dca8872281b3c47705e21217c8bc63912d95c9e2a7cac6bdf"
+  end
+
+  resource "numpy" do
+    url "https://pypi.python.org/packages/e0/4c/515d7c4ac424ff38cc919f7099bf293dd064ba9a600e1e3835b3edefdb18/numpy-1.11.1.tar.gz"
+    sha256 "dc4082c43979cc856a2bf352a8297ea109ccb3244d783ae067eb2ee5b0d577cd"
+  end
+
+  patch :DATA if build.with? "libespeak"
```
Comment (tdsmith) [inline aeneas.rb]: Can this be submitted upstream?

File: aeneas.rb [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+  resource "lxml" do
+    url "https://pypi.python.org/packages/11/1b/fe6904151b37a0d6da6e60c13583945f8ce3eae8ebd0ec763ce546358947/lxml-3.6.0.tar.gz"
+    sha256 "9c74ca28a7f0c30dca8872281b3c47705e21217c8bc63912d95c9e2a7cac6bdf"
+  end
+
+  resource "numpy" do
+    url "https://pypi.python.org/packages/e0/4c/515d7c4ac424ff38cc919f7099bf293dd064ba9a600e1e3835b3edefdb18/numpy-1.11.1.tar.gz"
+    sha256 "dc4082c43979cc856a2bf352a8297ea109ccb3244d783ae067eb2ee5b0d577cd"
+  end
```
Comment (tdsmith) [inline aeneas.rb]: Depend on `homebrew/python/numpy` instead.

File: aeneas.rb [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+
+  patch :DATA if build.with? "libespeak"
+
+  def install
+    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
+    resources.each do |r|
+      r.stage do
+        system "python", *Language::Python.setup_install_args(libexec/"vendor")
+      end
+    end
```
Comment (tdsmith) [inline aeneas.rb]: See the `numpy` formula for how to depend on an additional python module; you should drop a .pth file into the site-packages under `lib`.

File: aeneas.rb (hunk 4 above)
Comment (danielbair) [inline aeneas.rb]: I'm working on this here... https://github.com/Homebrew/homebrew-core/pull/2726

File: aeneas.rb (hunk 8 above)
Comment (danielbair) [inline aeneas.rb]: Ok, I will do that. Should I also create unique formula for lxml and beautifulsoup4?

File: aeneas.rb (hunk 7 above)
Comment (danielbair) [inline aeneas.rb]: I'm working with the upstream developer here... https://github.com/readbeyond/aeneas/pull/90

File: aeneas.rb (hunk 5 above)
Comment (danielbair) [inline aeneas.rb]: Ok, I patterned this after the caveats in the numpy.rb formula.

File: aeneas.rb (hunk 4 above)
Comment (danielbair) [inline aeneas.rb]: Ok, homebrew-core has accepted the updated espeak formula with libespeak static and dynamic formulas. See https://github.com/Homebrew/homebrew-core/commit/82a92f34c8130587e3c8359e5b034983b72dc65e

**Comment by danielbair** [comment]
@tdsmith would you be able to review this pull request again. I have attempted to implement the changes you requested. I have also been working with the developer and with the latest aeneas v1.5.1.0 release I am now able to eliminate the patches. I have also split out lxml.rb and bs4.rb to be used along with the existing numpy formula as dependencies.

**Comment by tdsmith** [comment]
Please vendor BeautifulSoup and lxml as `resource`s on aenas instead of submitting separate formulas.

File: aeneas.rb [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+      user_site_packages = Language::Python.user_site_packages "python"
+      <<-EOS.undent
+        If you use system python (that comes - depending on the OS X version -
+        with older versions of numpy, scipy and matplotlib), you may need to
+        ensure that the brewed packages come earlier in Python's sys.path with:
+          mkdir -p #{user_site_packages}
+          echo 'import sys; sys.path.insert(1, "#{homebrew_site_packages}")' >> #{user_site_packages}/homebrew.pth
+      EOS
+    end
+  end
```
Comment (tdsmith) [inline aeneas.rb]: Please remove these caveats.

**Comment by danielbair** [comment]
Ok, caveats removed, and vendored beautifulsoup4 and lxml as resources for aeneas formula.

**Comment by danielbair** [comment]
@tdsmith if this is acceptable, would you be able to merge this pull request as @pettarin has already included this homebrew formula in his instructions for installing aeneas on Mac (https://github.com/readbeyond/aeneas/blob/master/wiki/INSTALL.md).

**Comment by tdsmith** [comment]
@BrewTestBot test this please

**Comment by tdsmith** [comment]
This looks good, sorry for the delay. I'll make some minor changes when I pull it.

**Comment by tdsmith** [comment]
(Thanks for your patience!)

**Pull request closed without merging**